### ISSUE TITLE
Cannot write data to Cloudstor / NFS / EFS volumes in 18.09.2 Template

### ISSUE THREAD

**Issue by paullj1** [docker/for-aws#196]
### Expected behavior
* Should be able to write data to a Cloudstor volume mounted inside a container

### Actual behavior
* Any attempt to write data (perhaps only binary data?) to the Cloudstor volume hangs the container indefinitely

### Information
* This is sort of a duplicate of #177, but wanted to document the fact that it affects the current template as well.

### Steps to reproduce the behavior
  1. Create a brand new stack using the latest stable template
  2. SSH to any of the manager instances
  3. Create a Cloudstor volume
  4. Start a container with that Cloudstor volume mapped somewhere inside
  5. Run: `echo 'test' > /path/to/vol/test_file` (this should work)
  6. Run: `cat /path/to/vol/test_file` (this should work)
  7. Run: `dd if=/dev/urandom of=/path/to/vol/test_file2 bs=1M count=10` (this will hang indefinitely)

**Comment by paullj1** [comment]
@stevekerrison

**Comment by stevekerrison** [comment]
I emailed Docker support last week asking if these 'for AWS' projects had been abandoned, or what their status was. I feel, given that 19.03 engine is out now, this is particularly timely, as for the 18.09.2 release, Docker for AWS was updated some 13 days later. I haven't had a reply of any kind yet, so have just probed the ticket again.

**Comment by IJOL** [comment]
Feeling the same pain, just after reading this, I wonder  me trying to do it myself ( botstraping a regular swarm in aws ) .. using regular ubuntu instances is not the real way to go, i add that docker engine itself is very hung, i did  

~ $ docker volume create --driver cloudstor:aws --opt size=20 --opt backing=shared moodle_code
moodle_code
~ $ docker volume ls
DRIVER              VOLUME NAME
cloudstor:aws       moodle_code
local               sshkey
~ $ docker volume create --driver cloudstor:aws --opt size=20 --opt backing=relocatable --opt ebstype=gp2 p
ortainer_portainer_data
portainer_portainer_data
~ $ docker volume ls
DRIVER              VOLUME NAME
cloudstor:aws       moodle_code
cloudstor:aws       portainer_portainer_data
local               sshkey
~ $ docker volume inspect moodle_code
[
    {
        "CreatedAt": "0001-01-01T00:00:00Z",
        "Driver": "cloudstor:aws",
        "Labels": {},
        "Mountpoint": "/mnt/efs/reg/moodle_code",
        "Name": "moodle_code",
        "Options": {
            "backing": "shared",
            "size": "20"
        },
        "Scope": "local"
    }
]

 I know of the bogun size option in the first volume, it was not there the first time i triied so not a problems itself, after tryin to generate a simple ubuntu docker run using the volume and tryin to upload something in it makes the whoe system hung foreever doing a volume ls for example

**Comment by IJOL** [comment]
Just to let know to everyone thats ends here using the template in https://editions-us-east-1.s3.amazonaws.com/aws/stable/18.03.0/Docker.tmpl you will get a swarm cluster where efs volumes work ok

**Comment by paullj1** [comment]
Indeed, but the 18.03 Docker engine has critical CVEs published against it.
